resources/projectsyncup-design-document.md

```diff
@@ -0,0 +1,199 @@
+# Design Document
+
+## ProjectSyncUp Design
+
+## 1. Problem Statement
+
+_The objective of this project is to create a web application using Java, that helps streamline collaboration among team members working on projects together. The app will allow managers to create projects and assign tasks, while team members can track, prioritize and update progress of their assigned tasks, resulting in an efficient tool that increases project transparency for teams while working together._
+
+## 2. Project Scope
+
+### 3. Use Cases
+
+U1. As a manager, I want to create a new, empty project with a given name.
+
+U2. As a manager, I want to create a new, empty task with a given task name, description, due date, and list of assigned team members.
+
+U3. As a manager, I want to add a task, to a project with a given name.
+
+U4. As a manager, I want to update a task with different information.
+
+U5. As a manager, I want to delete a task, that has been added to a project.
+
+U6. As a manager, I want to view all projects I have created.
+
+U7. As a team member, I want to view all tasks assigned to me.
+
+U8. As a team member, I want to view all tasks assigned to me in a prioritized order (due date).
+
+U9. As a team member, I want to be able to update the progress of specific task (Not Started, In Progress, Complete).
+
+### 4. Stretch Goals
+U10. As a team member, I want to be able to track the amount of time I spend on each task in order to accurately bill for my work or manage my own productivity.
+
+U11. Admin can set roles for a user to be a team member or manager. (Multiple roles)
+
+U12. As a team member, I want to be able to comment on a specific task I am assigned to.
+
+U13. As a manager, I want to be able to share and collaborate files and documents with my team.
+
+
+## 5. Proposed Architecture Overview
+
+_Describe broadly how you are proposing to solve for the requirements you described in Section 2. This may include class diagram(s) showing what components you are planning to build. You should argue why this architecture (organization of components) is reasonable. That is, why it represents a good data flow and a good separation of concerns. Where applicable, argue why this architecture satisfies the stated requirements._
```

Comment:
Don't forget to scrub out the boilerplate instructions like this throughout the doc.